File: Swift/프로그래머스/0/120863. 다항식 더하기/다항식 더하기.swift
import Foundation

func solution(_ polynomial:String) -> String {
    var coefficientX = 0
    var constant = 0
    
    let elements = polynomial.split(separator: " ")
    
    for element in elements {
        if element.contains("x") {
            if element == "x" {
                coefficientX += 1
            } else {
                if let value = Int(element.dropLast()) {
                    coefficientX += value
                }
            }
        } else if let value = Int(element) {
            constant += value
        }
    }
    
    switch (coefficientX, constant) {
        case (0, 0):
            return "0"
        case (0, _):
            return "\(constant)"
        case (1, 0):
            return "x"
        case (1, _):
            return "x + \(constant)"
        case (_, 0):
            return "\(coefficientX)x"
        default:
            return "\(coefficientX)x + \(constant)"
    }
}

File: Python3/프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    answer = [0, 0]
    elements = polynomial.split(" ")
    
    for element in elements:
        if "x" in element:
            if len(element) == 1:
                answer[0] += 1
                continue
            answer[0] += int(element[0:len(element) - 1])
        elif element == "+":
            continue
        else:
            answer[1] += int(element[0:len(element)])
    
    x_part = f"{answer[0]}x" if answer[0] != 1 else "x"
    if answer[0] == 0:
        return str(answer[1])
    elif answer[1] == 0:
        return x_part
    else:
        return f"{x_part} + {answer[1]}"